error: conflicting types for ‘scanhash_asm32’

Hi.

I am trying to compile `cpuminer` on my Debian unstable system (amd64 kernel, i386 userland) and I have a problem with conflicting types for `scanhash_asm32`:

(...)
gcc -DHAVE_CONFIG_H -I. -pthread -fno-strict-aliasing    -DHAS_YASM -Ofast -Wall -march=native -MT minerd-sha256_cryptopp.o -MD -MP -MF .deps/minerd-sha256_cryptopp.Tpo -c -o minerd-sha256_cryptopp.o `test -f 'sha256_cryptopp.c' || echo './'`sha256_cryptopp.c
sha256_cryptopp.c:582:7: error: conflicting types for ‘scanhash_asm32’
In file included from sha256_cryptopp.c:9:0:
miner.h:166:14: note: previous declaration of ‘scanhash_asm32’ was here
make[2]: **\* [minerd-sha256_cryptopp.o] Error 1
make[2]: Leaving directory `/tmp/cpuminer'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory`/tmp/cpuminer'
make: **\* [all] Error 2

Upon inspecting, the code, indeed, there prototype in miner.h has an argument (`unsigned char *hash1`) that the code in function definition in sha256_cryptopp.c lacks.

--- a/cpu-miner.c
+++ b/cpu-miner.c
@@ -628,7 +628,7 @@ static void *miner_thread(void *userdata)
 #ifdef WANT_CRYPTOPP_ASM32
 		case ALGO_CRYPTOPP_ASM32:
 			rc = scanhash_asm32(thr_id, work.midstate, work.data + 64,
-				        work.hash1, work.hash, work.target,
+				        work.hash, work.target,
 					max_nonce, &hashes_done);
 			break;
 #endif

--- a/miner.h
+++ b/miner.h
@@ -164,7 +164,7 @@ extern bool scanhash_cryptopp(int, const unsigned char *midstate,unsigned char *
 	      unsigned char *hash, const unsigned char *target,
 	      uint32_t max_nonce, unsigned long *hashes_done);
 extern bool scanhash_asm32(int, const unsigned char *midstate,unsigned char *data,
-	      unsigned char *hash1, unsigned char *hash,
+	      unsigned char *hash,
 	      const unsigned char *target,
 	      uint32_t max_nonce, unsigned long *hashes_done);
 extern int scanhash_sse2_64(int, const unsigned char *pmidstate, unsigned char *pdata,

